### PR TITLE
Add hardware support for multiple functions on s390x.

### PR DESCRIPTION
**Pull request by stliibm**
This patch-series is adding hardware support on s390x for functions like fpclassify, round, ... - please have a look at the list of the patches.

For all of those functions, I've either enhanced the tests or created new tests if needed.
While adding new tests, I've also found and fixed some fails in the implementation of ieee754r-version - e.g. fdim - or powerpc-version - e.g. fpclassify.
Those fails are also commented in the corresponding commit-message.

I've tested this patch-series on s390x/s390, powerpc, x86_64.
There are no new test-failures.

Those patches are also applying on top of my already existing pull-request #72 "Fix some issues on s390x (see referenced issue for each commit)".

### CODE REVIEW

**Comment by stliibm** [comment]
I've just rebased the branch hw-support-on-s390x-0 in stliibm/libdfp to the latest upstream commits (pull-request #72 "Fix some issues on s390x (see referenced issue for each commit)" was merged) and re-build / -tested it. No changes are required.

File: sysdeps/s390/dfpu/samequantumd32.c [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  _Decimal64 tmpy = (_Decimal64) y;
+#endif
+
+  __asm__ ("lhi %0,1    \n\t"  /* Same exponent.  */
+#if _DECIMAL_SIZE == 128
+	   "cextr %1,%2 \n\t"
+#else
+	   "cedtr %1,%2 \n\t"
+#endif
+	   "je 1f       \n\t"  /* Jump away if exponents are equal. */
```
Comment (racardoso) [inline sysdeps/s390/dfpu/samequantumd32.c]: must have two spaces after the point here.

File: sysdeps/s390/dfpu/issignalingd32.c [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# define FUNCTION_NAME issignaling
+#endif
+#include <dfpmacro.h>
+
+int
+INTERNAL_FUNCTION_NAME (DEC_TYPE x)
+{
+  int result;
+  __asm__ ("lhi %0,1   \n\t" /* Is signaling nan.  */
+	   /* Is data class signaling nan?  */
```
Comment (racardoso) [inline sysdeps/s390/dfpu/issignalingd32.c]: this comment seems misaligned.

File: sysdeps/s390/dfpu/isnormald32.c [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+# define FUNCTION_NAME isnormal
+#endif
+#include <dfpmacro.h>
+
+int
+INTERNAL_FUNCTION_NAME (DEC_TYPE x)
+{
+  int result;
+  __asm__ ("lhi %0,0     \n\t" /* Is not normal.  */
+	   /* Is data class normal?  */
```
Comment (racardoso) [inline sysdeps/s390/dfpu/isnormald32.c]: This comment seems misaligned.

File: sysdeps/s390/dfpu/signbitd32.c [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+# define FUNCTION_NAME signbit
+#endif
+#include <dfpmacro.h>
+
+int
+INTERNAL_FUNCTION_NAME (DEC_TYPE x)
+{
+  int result;
+  __asm__ ("lhi %0,1      \n\t" /* Is negative.  */
+	   /* Is -zero, -subnormal, -inf, -nan?  */
```
Comment (racardoso) [inline sysdeps/s390/dfpu/signbitd32.c]: This comment also seems misaligned.

File: tests/fmin.input [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+0.0	0.0		0.0
+2.0	1.0		1.0
+1.0	2.0		1.0
+2.0	2.0		2.0
+-1.0	-2.0		-2.0
+-2.0	-1.0		-2.0
+-2.0	-2.0		-2.0
+2.123	1.86743		1.86743
+1.86744	1.86743		1.86743
```
Comment (racardoso) [inline tests/fmin.input]: Many of those .input tests have extra spaces/tabs between the column. I am not sure about the code standards here but I guess we should not have those spaces here. But, I am not sure about it. Maybe @gftg85 or @tuliom  have a opinion about it.

**Comment by stliibm** [comment]
Hi Rogerio,
thank you for the review. I've added the extra space in the comment and pushed the commit "Fix commit in samequantumd32.c".


Regarding the misaligned comments:
I've assumed that we use a tab width of 8 characters as used in glibc?
Then the comments are starting after '  __asm__ (' and are aligned in the same way as the instructions.

Regarding input .input tests:
Here is the same, I've assumed a tab width of 8 characters.
Then the arguments and return values are all in columns.

Please let me know if I've assumed the false tab width and/or if there are news about the code standards with spaces/tabs in .input files.

Thanks,
Stefan

**Comment by racardoso** [comment]
@stliibm Hi Stephan, no you're right. I made two small mistakes here. I've checked the files on Github and in my vim but, my vim was with the wrong TAB/Space configuration at the time so in both Github view and vim it seems misaligned. But, I've checked it here again and it's not.

In fact you can assume that we are use the same code standards as glibc. Although we need to define it formally on Contribute doc (Code Standard has a TODO there). There's some places on the code with typo's, trailing spaces, extra tabs and, indentation that must be fix on the code.

**Pull request closed without merging**